### DISCUSSION TITLE
MicroPython and Wireguard

### DISCUSSION THREAD

**Discussion by moferreira** [micropython/micropython#15966] [ESP32] [answered]
Does MicroPython implements Wireguard? If so, can you share how to configure it?

**Comment by bidrohini** [comment]
Looks like it does not: https://forum.micropython.org/viewtopic.php?t=7569

**Comment by Josverl** [comment, marked as answer]
Wireguard by design needs a basic OS and IP stack.
There is a project to run it on esp32 RTOS https://github.com/trombik/esp_wireguard

But not aware of integration of that with micropython

**Reply by hisacro** [reply]
> Wireguard by design needs a basic OS and IP stack. There is a project to run it on esp32 RTOS https://github.com/trombik/esp_wireguard
> 
> But not aware of integration of that with micropython

Things have changed in 2026, https://docs.arduino.cc/libraries/wireguard-esp32/

**Comment by sosi-deadeye** [comment]
Here is a possible solution with an embedded PC: https://milkv.io/duo
Micropython doesn't run on this hardware bar metal, you could run it as the Unix version.
